lab_1_classify_by_unigrams/main.py

```diff
@@ -6,55 +6,103 @@
+    return frequencies
 
 
 def create_language_profile(language: str, text: str) -> dict[str, str | dict[str, float]] | None:
     """
-    Creates a language profile
-    :param language: a language
-    :param text: a text
-    :return: a dictionary with two keys – name, freq
-    """
+        Creates a language profile
+        :param language: a language
+        :param text: a text
+        :return: a dictionary with two keys – name, freq
+        """
+    if not isinstance(language, str) or not isinstance(text, str):
+        return None
+    freq_dict = calculate_frequencies(tokenize(text))
+    if not freq_dict:
+        return None
+    return {'name': language, 'freq': freq_dict}
 
 
 def calculate_mse(predicted: list, actual: list) -> float | None:
     """
-    Calculates mean squared error between predicted and actual values
-    :param predicted: a list of predicted values
-    :param actual: a list of actual values
-    :return: the score
-    """
+        Calculates mean squared error between predicted and actual values
+        :param predicted: a list of predicted values
+        :param actual: a list of actual values
+        :return: the score
+        """
+    if (not isinstance(predicted, list)
+            or not isinstance(actual, list)
+            or len(predicted) != len(actual)):
+        return None
+    total = len(predicted)
+    score = 0
+    for i in range(total):
+        score += (actual[i] - predicted[i]) ** 2 / total
+    return score
 
 
 def compare_profiles(
         unknown_profile: dict[str, str | dict[str, float]],
         profile_to_compare: dict[str, str | dict[str, float]]
 ) -> float | None:
     """
-    Compares profiles and calculates the distance using symbols
-    :param unknown_profile: a dictionary of an unknown profile
-    :param profile_to_compare: a dictionary of a profile to compare the unknown profile to
-    :return: the distance between the profiles
-    """
+        Compares profiles and calculates the distance using symbols
+        :param unknown_profile: a dictionary of an unknown profile
+        :param profile_to_compare: a dictionary of a profile to compare the unknown profile to
+        :return: the distance between the profiles
+        """
+    target_set = {'freq', 'name'}
+    if (not isinstance(unknown_profile, dict)
+            or not isinstance(profile_to_compare, dict)
+            or len(target_set & set(unknown_profile) & set(profile_to_compare)) != len(target_set)):
+        return None
+    new_freq = {a: .0 for a in set(list(unknown_profile['freq']) +
+                                   list(profile_to_compare['freq']))}
+    freq_to_compare = new_freq.copy()
+    assert isinstance(unknown_profile['freq'], dict)
+    new_freq.update(unknown_profile['freq'])
+    assert isinstance(profile_to_compare['freq'], dict)
+    freq_to_compare.update(profile_to_compare['freq'])
+    distance = calculate_mse(list(new_freq.values()), list(freq_to_compare.values()))
```

Comment:
do not create the variable - return `dict` as is

Reply:
omitted the 'distance' variable